src/server/replica.cc

```diff
@@ -693,11 +693,16 @@ error_code Replica::ConsumeRedisStream() {
   };
   RETURN_ON_ERR(exec_st_.SwitchErrorHandler(std::move(err_handler)));
 
-  facade::CmdArgVec args_vector;
+  CmdArgVec args_vector;
 
   acks_fb_ = fb2::Fiber("redis_acks", &Replica::RedisStreamAcksFb, this);
 
   while (true) {
+    // Yield if the fiber has been running for long.
+    if (base::CycleClock::ToUsec(ThisFiber::GetRunningTimeCycles()) > 1000) {  // 1ms
+      ThisFiber::Yield();
+    }
```

Comment:
Its not clear what tests to add for this, perhaps https://github.com/dragonflydb/dragonfly/issues/4451 not failing (or continuing to fail) is a test